### ISSUE TITLE
Extend TryGenerateExpectedSourceFileAndGetDiffLink to cover all tests in a file

### ISSUE THREAD

**Issue by alrz** [dotnet/roslyn#26161]
I found this method in the test suite: `TryGenerateExpectedSourceFileAndGetDiffLink`

It generates the whole source file with just the expected assertion string replaced with actual.

It has some limitations and cannot be used for bulk baseline changes,

- It only works if a diff tool is available
- It only works for a single test

The most needed methods that need to have this is `VerifyIL` and `VerifyDiagnostics` mainly in case of bulk baseline changes.

The idea is that to keep a single file and apply new changes to that file so the expected source file contains changes for all tests in the file.

However, in order to do that, we should keep track of line numbers because they might get out of date once a test expected string is updated in the source.

Also, since `VerifyDiagnostics` accepts a params, we can't have line number at all. maybe fallback to stackframe? (my motivating use case for this feature request https://github.com/dotnet/csharplang/issues/1458)

Thoughts?

/cc @tmat

**Comment by jcouv** [comment]
I didn't fully understand the problem you're trying to address. Could you clarify?

Assuming it is "how can we update the baseline for multiple tests quickly?", here are some thoughts:
In the past. I've proposed a small mitigation: when a test fails, the actual result could be copied to the clipboard.
There is a more advanced variant: when a test fails, the actual result is saved to a file, indexed by the name of the test and position within that test. A code fixer can read that file and offer to update the baseline. (The advantage of this approach is that the system clipboard is left untouched)
I have a [prototype for the first variant](https://github.com/jcouv/roslyn/commit/776e2eb3ced6d5dd05077f77946e797fff9a3759) if you're interested.

**Comment by alrz** [comment]
Yeah I saw the clipboard solution, but I really want to have this for the whole file in one go.

>A code fixer can read that file and offer to update the baseline. 

Assuming we could have a fix-all I think this'd worth to invest on. For VerifyDiagnostics however, we can't get the exact line number, though it doesn't seem to be hard to workaround.

**Comment by jcouv** [comment]
In last discussion, the team pushed back on the idea of a FixAll. Fixing tests one by one is slower but also affords more scrutiny....

**Comment by alrz** [comment]
Unless you review the git diff before pushing changes, which you do anyways. Current copy/paste experience does really no good in that IMO.
